### PR TITLE
wasser: add ubuntu config

### PR DESCRIPTION
**Pull request by kshtsk**
Multiple fixes in order to make ubuntu development host testable.

Signed-off-by: Kyr Shatskyy <kyrylo.shatskyy@suse.com>

## Checklist
- [ ] References issues, create if required
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins test tumbleweed`
- `jenkins run tumbleweed`
- `jenkins test leap`
- `jenkins run leap`

</details>

### CODE REVIEW

**Review by AvengerMoJo — COMMENTED** [review]
update the setup script to remove the wasser setup

**Comment by kshtsk** [comment]
jenkins test ubuntu

**Comment by kshtsk** [comment]
jenkins run opensuse

**Comment by kshtsk** [comment]
jenkins run ubuntu

**Comment by kshtsk** [comment]
jenkins run ubuntu

**Comment by kshtsk** [comment]
jenkins run ubuntu

**Comment by kshtsk** [comment]
jenkins run ubuntu

**Comment by kshtsk** [comment]
jenkins run ubuntu

**Comment by kshtsk** [comment]
jenkins run ubuntu

**Comment by kshtsk** [comment]
The test results are false positive:
```
02:02:10 +++  ./tools/aqrdev create foo
02:02:10 >>> Importing image 'aquarium' as Vagrant box
02:02:12 >>> Imported image 'aquarium' as Vagrant box
02:02:23 >>> success!
02:02:24 ||| exit code: 0
02:02:24 === start deployment
02:02:24 +++  cd aquarium
02:02:24 +++  source venv/bin/activate
02:02:24 +++  ./tools/aqrdev start foo --conservative
02:02:24 >>> Bringing machine 'node1' up with 'libvirt' provider...
02:02:32 >>> Bringing machine 'node2' up with 'libvirt' provider...
02:02:32 EEE Error while connecting to Libvirt: Error making a connection to libvirt URI qemu:///system?no_verify=1&keyfile=/home/ubuntu/.ssh/id_rsa:
02:02:33 EEE Call to virConnectOpen failed: Failed to connect socket to '/var/run/libvirt/libvirt-sock': Permission denied
02:02:33 >>> Deployment 'foo' started
02:02:33 ||| exit code: 0
02:02:33 === dump aquarium service log
02:02:33 +++  cd aquarium/tests/vagrant/deployments/foo
02:02:33 +++  vagrant ssh -c "sudo journalctl -u aquarium"
02:02:33 >>> ==> node1: Domain is not created. Please run `vagrant up` first.
02:02:36 ||| exit code: 0
02:02:36 === check aquarium service is up and running
02:02:36 +++  cd aquarium/tests/vagrant/deployments/foo
02:02:36 +++  vagrant ssh -c "sudo systemctl status aquarium"
02:02:36 >>> ==> node1: Domain is not created. Please run `vagrant up` first.
02:02:39 ||| exit code: 0
```

First problem is that `aqrdev start` cannot connect to qemu, however it reports that deployment is started and returns 0 exit code.
```
02:02:24 +++  ./tools/aqrdev start foo --conservative
02:02:24 >>> Bringing machine 'node1' up with 'libvirt' provider...
02:02:32 >>> Bringing machine 'node2' up with 'libvirt' provider...
02:02:32 EEE Error while connecting to Libvirt: Error making a connection to libvirt URI qemu:///system?no_verify=1&keyfile=/home/ubuntu/.ssh/id_rsa:
02:02:33 EEE Call to virConnectOpen failed: Failed to connect socket to '/var/run/libvirt/libvirt-sock': Permission denied
02:02:33 >>> Deployment 'foo' started
02:02:33 ||| exit code: 0
```

Another problem `vagrant ssh` fails to ssh node1, however it returns exit code 0.

**Comment by kshtsk** [comment]
jenkins run ubuntu

**Comment by kshtsk** [comment]
jenkins run tumbleweed

File: tools/setup-dev.sh [hunk 1]
```diff
@@ -227,6 +227,13 @@ if ! ${skip_install_deps} ; then
         echo "Dependency installation failed"
         exit 1
       }
+      echo "=> make sure user in libvirt and starting libvirtd"
+      USER=`whoami`
+      sudo usermod -a -G libvirt $USER
+      sudo mv /etc/libvirt/libvirtd.conf /etc/libvirt/libvirtd.conf.bak
+      sudo sed -e 's/#unix_sock_\([g|r]\)/unix_sock_\1/' /etc/libvirt/libvirtd.conf.bak > /etc/libvirt/libvirtd.conf
+      sudo systemctl restart libvirtd
+
```
Comment (kshtsk) [inline tools/setup-dev.sh:236]: Looks like this does not change anything

Comment (AvengerMoJo) [reply tools/setup-dev.sh:236]: I just finished a fresh install and it seems to be working fine for me

Comment (kshtsk) [reply tools/setup-dev.sh:236]: see the logs, it is not working on remote machines, same as before...

**Comment by kshtsk** [comment]
jenkins test ubuntu

File: tools/setup-dev.sh (hunk 1 above)
Comment (kshtsk) [reply tools/setup-dev.sh]: this code is breaking automation now, it just hangs after the 'echo' message.

Comment (kshtsk) [reply tools/setup-dev.sh]: If I run this manually and press Ctrl+D I'm getting this:
```
./tools/setup-dev.sh: line 236: /etc/libvirt/libvirtd.conf: Permission denied
Submodule 'ceph.git' (https://github.com/aquarist-labs/ceph.git) registered for path 'src/gravel/ceph.git'
Cloning into '/home/ubuntu/aquarium/src/gravel/ceph.git'...
```

File: tools/setup-dev.sh [hunk 2]
```diff
@@ -227,6 +228,14 @@ if ! ${skip_install_deps} ; then
         echo "Dependency installation failed"
         exit 1
       }
+      echo "=> make sure user in libvirt and starting libvirtd"
+      USER=`whoami`
+      sudo usermod -a -G libvirt $USER
+      newgrp libvirt
+      sudo mv /etc/libvirt/libvirtd.conf /etc/libvirt/libvirtd.conf.bak
+      sudo sed -e 's/#unix_sock_\([g|r]\)/unix_sock_\1/' /etc/libvirt/libvirtd.conf.bak > /etc/libvirt/libvirtd.conf
```
Comment (kshtsk) [inline tools/setup-dev.sh:236]: I guess this is problematic:
```
$ sudo sed -e 's/#unix_sock_\([g|r]\)/unix_sock_\1/' /etc/libvirt/libvirtd.conf.bak > /etc/libvirt/libvirtd.conf
-bash: /etc/libvirt/libvirtd.conf: Permission denied
```
It is because you're still ubuntu user when redirecting stdout.

File: tools/setup-dev.sh [hunk 3]
```diff
@@ -227,6 +228,14 @@ if ! ${skip_install_deps} ; then
         echo "Dependency installation failed"
         exit 1
       }
+      echo "=> make sure user in libvirt and starting libvirtd"
+      USER=`whoami`
```
Comment (kshtsk) [inline tools/setup-dev.sh:232]: Really? Why do you override USER environment?

**Comment by kshtsk** [comment]
jenkins test ubuntu

**Comment by kshtsk** [comment]
jenkins test ubuntu

**Comment by kshtsk** [comment]
jenkins test ubuntu

**Comment by kshtsk** [comment]
jenkins test ubuntu

**Comment by kshtsk** [comment]
jenkins test ubuntu

**Comment by kshtsk** [comment]
finally I've got it working

**Comment by kshtsk** [comment]
@jecluis could please take a look in particular on `tools/libaqr/images: simplify _check_for_image` (14558a18)

File: tools/setup-dev.sh (hunk 1 above)
Comment (kshtsk) [reply tools/setup-dev.sh]: I dropped your overall changes because it appears they are irrelevant to the issue, the real reason for these errors:
```
02:02:24 +++  ./tools/aqrdev start foo --conservative
02:02:24 >>> Bringing machine 'node1' up with 'libvirt' provider...
02:02:32 >>> Bringing machine 'node2' up with 'libvirt' provider...
02:02:32 EEE Error while connecting to Libvirt: Error making a connection to libvirt URI qemu:///system?no_verify=1&keyfile=/home/ubuntu/.ssh/id_rsa:
02:02:33 EEE Call to virConnectOpen failed: Failed to connect socket to '/var/run/libvirt/libvirt-sock': Permission denied
```

is that the old ssh connection was remaining for the user as it was before libvirt setup, and reconnecting the ssh after setup-dev.sh resolves the problem.

**Comment by kshtsk** [comment]
jenkins test ubuntu

**Comment by kshtsk** [comment]
jenkins run tumbleweed

File: tools/libaqr/images.py [hunk 4]
```diff
@@ -108,7 +108,7 @@ def _check_for_image(cls, path: Path) -> Tuple[str, Path]:
             elif imgpath.suffix == ".box" and entry.count("vagrant") > 0:
                 img = imgpath
 
-        if not raw_img or not img:
+        if not raw_img and not img:
```
Comment (jecluis) [inline tools/libaqr/images.py]: This will not raise if `img` is missing, as if it's everything just fine. I think the bigger question is: when is it possible for .box to be present, but not .raw? If there's a reasonable expectation that it might be the case without it being an error somewhere else, then the check should check for `not img` standalone.

TL;DR: this patch introduces a regression.

**Review by jecluis — CHANGES_REQUESTED** [review]
I'd drop the intermediate patch and keep just `96c85a0894e9d1e1ec1fcbeda996d6ead76ec760`. The latter looks fine, but the other one introduces a regression, and is essentially overwritten by the latter anyway.

File: tools/libaqr/images.py (hunk 4 above)
Comment (kshtsk) [reply tools/libaqr/images.py]: Gonna stash it, as for when it is possible .raw not present, then this is exactly the case when we run it on Ubuntu.

**Comment by kshtsk** [comment]
jenkins test ubuntu

**Review by jecluis — APPROVED** [review]
